api/client/client.cc

```diff
@@ -106,23 +112,47 @@ int main(int argc, char **argv) {
         Help();
         return 1;
     }
-
-    string cmd = string(argv[1]);
-    if (cmd == "nic") {
-        return SubNic(argc, argv, options);
-    } else if (cmd == "sg") {
-        return SubSg(argc, argv, options);
-    } else if (cmd == "status") {
-        return sub_status(argc, argv, options);
-    } else if (cmd == "shutdown") {
-        return sub_shutdown(argc, argv, options);
-    } else if (cmd == "request") {
-        return SubRequest(argc, argv, options);
-    } else if (cmd == "dump") {
-        return SubDump(argc, argv, options);
-    } else {
-        cerr << "invalid subcommand " << cmd << endl;
-        Help();
-        return 1;
+    int i = 0;
+    int id_command = 0;
+    int id_insert = 0;
+    std::vector<Cmd> cmds;
+    Cmd command = {argv, 0};
+    cmds.push_back(command);
+    while (i < argc) {
+        string cmd = argv[id_insert];
+        if (cmd == "+") {
+            cmds.at(id_command).size = id_insert;
+            argv += id_insert;
+            id_command++;
+            Cmd command = {argv, 0};
+            cmds.push_back(command);
+            id_insert = 1;
+        } else {
+            id_insert++;
+        }
+        i++;
+    }
+    cmds.at(id_command).size = id_insert;
+    int ret = 0;
```

Comment:
sorry, but your error handling is broken:
what you are doing here is ignoring every return but the last one, then retuning it.

I think you should not use ret here, but just exit at the first error, and return 0, at the end of the function.

Reply:
Still broken....